### PR TITLE
chore: add clarifying note for composite and expand term

### PR DESCRIPTION
**Pull request by JoviDeCroock**
Resolves https://github.com/graphql/graphql-spec/issues/1076

This PR adds a clarifying note for the term composite in prior versions of the spec and expands the term into `Object, Interface or Union type` wherever it was found

### CODE REVIEW

File: spec/Section 4 -- Introspection.md [hunk 1]
```diff
@@ -352,7 +352,7 @@ Fields\:
 
 **Input Object**
 
-Input objects are composite types defined as a list of named input values. They
+Input objects are types defined as a list of named input values. They
```
Comment (benjie) [inline spec/Section 4 -- Introspection.md:355]: I think the `composite` improves the clarity of the text. I think that the term "composite type" is fairly well understood in prose, e.g. [Wikipedia: composite data type](https://en.wikipedia.org/wiki/Composite_data_type). I suggest we revert this change.

```suggestion
Input objects are composite types defined as a list of named input values. They
```

File: spec/Section 5 -- Validation.md [hunk 2]
```diff
@@ -910,7 +910,7 @@ fragment inlineNotExistingType on Dog {
 }
 ```
 
-#### Fragments on Composite Types
+#### Fragments on object, interface or union Types
```
Comment (benjie) [inline spec/Section 5 -- Validation.md:913]: See the [style guide on titles](https://github.com/graphql/graphql-spec/blob/main/STYLE_GUIDE.md#title-case):

```suggestion
#### Fragments on Object, Interface or Union Types
```

**Review by benjie — CHANGES_REQUESTED** [review]
I support the replacement of "composite types" in section 5 with more explicit language.

**Review by benjie — DISMISSED** [review]
I think this helps clarity :+1:

**Comment by benjie** [comment]
@graphql/tsc Can we get another approval, then we'll wait a week or two for any more feedback and if no-one has concerns I think we should merge :+1:

**Comment by leebyron** [comment]
I’m OK with this change however I’m pretty sure the term composite type
shows up all over the graphql-js repository as well

**Review by Keweiqu — APPROVED** [review]
Makes sense! Thanks for doing this

**Comment by JoviDeCroock** [comment]
As suggested in yesterdays WG this has been replaced with a note about composite

**Comment by martinbonnin** [comment]
I was thinking we would keep "composite" out of the normative sections (initial PR) and add a non-normative note that "composite" may be found in other contexts:

```
Note: implementers may use the term composite type for a type that is either an object, interface
or union.
```

**Review by martinbonnin — APPROVED** [review]

**Comment by benjie** [comment]
@JoviDeCroock I suggest you undo your force-push and then add a note along the lines of what you just added; essentially the change should be to remove "composite" from most places (preferring explicitness) and then to add a non-normative note indicating that previous versions of the spec used and some implementers use the term "composite type" to refer to Object, Interface and Union types. 

Also, the letter after `Note: ` should be a capital since that is what is rendered in the note box (the `Note: ` prefix is stripped out).

File: spec/Section 5 -- Validation.md [hunk 3]
```diff
@@ -455,6 +455,9 @@ SameResponseShape(fieldA, fieldB):
   - If {SameResponseShape(subfieldA, subfieldB)} is {false}, return {false}.
 - Return {true}.
 
+Note: composite is used to signal a type that is either an object-, interface-
```
Comment (JoviDeCroock) [inline spec/Section 5 -- Validation.md:458]:
```suggestion
Note: Composite is used to signal a type that is either an object-, interface-
```

**Comment by JoviDeCroock** [comment]
@benjie the force push was just to update this branch with `main` I re-added the `composite` rephrasing and updated the note. Thank you for the review 🙌

File: spec/Section 5 -- Validation.md [hunk 4]
```diff
@@ -455,6 +455,9 @@ SameResponseShape(fieldA, fieldB):
   - If {SameResponseShape(subfieldA, subfieldB)} is {false}, return {false}.
 - Return {true}.
 
+Note: In prior versions of the spec the term "composite" wass used to signal
+a type that is either an object-, interface- or union-type.
```
Comment (benjie) [inline spec/Section 5 -- Validation.md:459]:
```suggestion
Note: In prior versions of the spec the term "composite" was used to signal
a type that is either an Object, Interface or Union type.
```

**Review by benjie — CHANGES_REQUESTED** [review]
Looking good!

**Review by martinbonnin — APPROVED** [review]
👍

File: spec/Section 5 -- Validation.md [hunk 5]
```diff
@@ -446,7 +446,7 @@ SameResponseShape(fieldA, fieldB):
 - If {typeA} or {typeB} is Scalar or Enum:
   - If {typeA} and {typeB} are the same type return {true}, otherwise return
     {false}.
-- Assert: {typeA} and {typeB} are both composite types.
+- Assert: {typeA} and {typeB} are both Object, Interface or Union types.
```
Comment (benjie) [inline spec/Section 5 -- Validation.md:449]: This reads ambiguously; it could be interpretted as `(typeA is object and typeB is object) or (typeA is union and typeB is union) or (typeA is an interface and typeB is an interface)` rather than `(typeA is object or union or interface) and (typeB is object or union or interface)`. Perhaps:

```suggestion
- Assert: {typeA} is an object, union or interface type.
- Assert: {typeB} is an object, union or interface type.
```

**Review by benjie — CHANGES_REQUESTED** [review]
This one got undone accidentally I think!

**Review by benjie — APPROVED** [review]
Looks great to me :+1:

**Comment by benjie** [comment]
I believe @Keweiqu and @leebyron's previous reviews are still valid with the latest state, though they may want to review the added note.

**Review by mjmahone — APPROVED** [review]
It's nice to have a word to use for "Types that are schema-defined vertex types in your graph" but Composite isn't a great word for that, agree.

**Review by robrichard — APPROVED** [review]
